src/main/java/tech/picnic/jolo/Entity.java

```diff
@@ -2,35 +2,34 @@
 
 import static java.util.Arrays.stream;
 import static java.util.stream.Stream.concat;
-import static tech.picnic.jolo.Util.equalFieldNames;
 import static tech.picnic.jolo.Util.validate;
 
-import java.util.Collection;
-import java.util.Collections;
-import java.util.LinkedHashMap;
-import java.util.Map;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Optional;
 import javax.annotation.Nullable;
 import org.jooq.Field;
 import org.jooq.Record;
 import org.jooq.Table;
 
 /**
- * Represents a mapping from a jOOQ table to a class. This class is used to store entities that have
- * been loaded from a data set.
+ * Represents a mapping from a {@link Table table} to a {@link Class class}. This class is used to
+ * load and instantiate objects from a data set.
  *
  * @param <T> The class that is mapped to.
  */
 public final class Entity<T, R extends Record> {
-  private final Map<Long, T> entities = new LinkedHashMap<>();
   private final Table<R> table;
   private final Field<Long> primaryKey;
   private final Class<T> type;
+  private final Field<?>[] fields;
+
+  private final int hashCode;
 
-  private Field<?>[] fields;
   @Nullable private Field<?>[] resultFields;
 
   /**
-   * Creates a mapping from a jOOQ table to the given class.
+   * Creates a mapping from a {@link Table table} to the given {@link Class class}.
```

Comment:
It seems like you want to make a code-base wide style change in the scope of a functional change. Personally I prefer the original. Note that the exact types are already clear from the signature, and that most developers will not feel like a link to the documentation of `Class` is very helpful. The context that it's the jOOQ table class is now missing so you'll have to do something extra (hover over the text?) to recover that information.